### DISCUSSION TITLE
vsxmake工程目录中没有添加的自定义文件类型

### DISCUSSION THREAD

**Discussion by Ligo04** [xmake-io/xmake#3904] [Q&A] [answered]
```
add_files("source/shaders/**.hlsl",{rule="hlsl_shader_complier"})
```

我定义了hlsl着色器文件的编译规则，并且使用 `xmake project -k vsxmake`

生成vsxmake工程，但是在vs的资源管理器目录中的并没有hlsl文件

![image-20230629220136278](https://img2023.cnblogs.com/blog/1656870/202306/1656870-20230629220245358-49943963.png)

我想知道有什么方案可以解决

**Comment by waruqi** [comment]
有些规则仅仅支持直接构建，不支持 project generator

**Reply by waruqi** [reply, marked as answer]
可以通过 add_headerfiles 添加，如果想要显示

**Comment by Ligo04** [comment]
> 可以通过 add_headerfiles 添加，如果想要显示

噢噢，可以了 ，哪路或多
